File: go/gin/internal/modules/account/module.go
package account

import (
	auth "tcc/internal/modules/auth/middleware"
	"tcc/internal/modules/branch"
	"tcc/internal/modules/client"
	"tcc/internal/modules/manager"

	"github.com/gin-gonic/gin"
)

func AccountModule(router *gin.Engine) {
	service := AccountService{
		repository: struct{account AccountRepository; client client.ClientRepository; branch branch.BranchRepository; manager manager.ManagerRepository}{
			account: AccountRepository{},
			client: client.ClientRepository{},
			branch: branch.BranchRepository{},
			manager: manager.ManagerRepository{},
		},
	}
	controller := AccountController{
		service: service,
	}

	group := router.Group("/contas")
	{
		group.GET("", auth.AuthMiddleware(), controller.FindAll)
		group.POST("", auth.AuthMiddleware(), controller.Create)
		group.GET("/:id", auth.AuthMiddleware(), controller.FindByCPF)
	}
}

File: rust/axum/src/modules/account/module.rs
use axum::{Router, routing::get, routing::post, middleware::from_fn,};
use super::controller::AccountController;
use crate::modules::auth::middleware::auth_middleware;

pub fn account_module() -> Router {
    let router = 
    Router::new()
    .route("/", get(AccountController::find_all))
    .layer(from_fn(auth_middleware))
    .route("/", post(AccountController::create))
    .layer(from_fn(auth_middleware))
    .route("/{id}", get(AccountController::find_by_id))
    .layer(from_fn(auth_middleware));

    Router::new().nest("/contas", router)
}
